sim/druid/claw.go

```diff
@@ -0,0 +1,80 @@
+package druid
+
+import (
+	"time"
+
+	"github.com/wowsims/classic/sim/core"
+)
+
+func (druid *Druid) registerClawSpell() {
+	flatDamageBonus := map[int32]float64{
+		20: 27,
+		28: 39,
+		38: 57,
+		48: 88,
+		60: 115,
+	}[druid.Level]
+
+	druid.Claw = druid.RegisterSpell(Cat, core.SpellConfig{
+		SpellCode: SpellCode_DruidClaw,
+		ActionID: core.ActionID{SpellID: map[int32]int32{
+			20: 1082,
+			28: 3029,
+			38: 5201,
+			48: 9849,
+			60: 9850,
+		}[druid.Level]},
+		SpellSchool: core.SpellSchoolPhysical,
+		DefenseType: core.DefenseTypeMelee,
+		ProcMask:    core.ProcMaskMeleeMHSpecial,
+		Flags:       core.SpellFlagMeleeMetrics | core.SpellFlagAPL | SpellFlagOmen | SpellFlagBuilder,
+
+		EnergyCost: core.EnergyCostOptions{
+			Cost:   45 - 1*float64(druid.Talents.Ferocity),
+			Refund: 0.8,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: time.Second,
+			},
+			IgnoreHaste: true,
+		},
+
+		DamageMultiplier: 1,
+		ThreatMultiplier: 1,
+		BonusCoefficient: 1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := flatDamageBonus + spell.Unit.MHWeaponDamage(sim, spell.MeleeAttackPower())
+
+			result := spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeMeleeSpecialHitAndCrit)
+
+			if result.Landed() {
+				druid.AddComboPoints(sim, 1, target, spell.ComboPointMetrics())
+			} else {
+				spell.IssueRefund(sim)
+			}
+		},
+		ExpectedInitialDamage: func(sim *core.Simulation, target *core.Unit, spell *core.Spell, _ bool) *core.SpellResult {
+			baseDamage := flatDamageBonus + spell.Unit.AutoAttacks.MH().CalculateAverageWeaponDamage(spell.MeleeAttackPower())
+
+			baseres := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeExpectedMagicAlwaysHit)
+
+			attackTable := spell.Unit.AttackTables[target.UnitIndex][spell.CastType]
+			critChance := spell.PhysicalCritChance(attackTable)
+			critMod := critChance * (spell.CritMultiplier(attackTable) - 1)
+
+			baseres.Damage *= 1 + critMod
```

Comment:
❓ what is this bit used for? Does claw do some pecial damage calculation?

Reply:
This was some leftover pasta from Shred, removed the expected damage bit entirely.